src/message/messageActionSheet.js

```diff
@@ -384,11 +380,42 @@ export const showActionSheet = (
   };
   showActionSheetWithOptions(
     {
-      ...(isHeader
-        ? {
-            title: getActionSheetTitle(params.message, params.backgroundData.ownUser),
-          }
-        : {}),
+      options: optionCodes.map(code => callbacks._(allButtons[code].title)),
+      cancelButtonIndex: optionCodes.length - 1,
+    },
+    callback,
+  );
+};
+
+export const showHeaderActionSheet = (
+  showActionSheetWithOptions: ShowActionSheetWithOptions,
+  callbacks: {|
+    dispatch: Dispatch,
+    startEditMessage: (editMessage: EditMessage) => void,
+    _: GetText,
+  |},
+  params: ConstructSheetParams<>,
+): void => {
+  const optionCodes = constructHeaderActionButtons(params);
+  const callback = buttonIndex => {
+    (async () => {
+      const pressedButton: ButtonDescription = allButtons[optionCodes[buttonIndex]];
+      try {
+        await pressedButton({
+          subscriptions: params.backgroundData.subscriptions,
+          auth: params.backgroundData.auth,
+          ownUser: params.backgroundData.ownUser,
+          ...params,
+          ...callbacks,
+        });
```

Comment:
I ended up actually keeping the duplication, since that turned out much easier than making the types work out once we had more than one type of button — that's worth the pretty small amount of duplication, IMO (although if there's a nice flow way to write it, I'm happy to see it)

Reply:
Sent #4566 with a way to do this. I agree the duplication isn't a big problem because the code is so small. But the `Alert.alert` bit in particular feels like the sort of thing that can easily get annoyingly out of sync when duplicated, so I think there's some value in keeping it deduplicated.